pkg/operator/controller/ingress/status.go

```diff
@@ -632,21 +633,41 @@ func formatConditions(conditions []*operatorv1.OperatorCondition) string {
 // computeIngressProgressingCondition computes the ingresscontroller's
 // "Progressing" status condition, which indicates if the ingresscontroller's
 // current state matches its desired state.
-func computeIngressProgressingCondition(conditions []operatorv1.OperatorCondition, ic *operatorv1.IngressController, service *corev1.Service, platform *configv1.PlatformStatus) operatorv1.OperatorCondition {
+func computeIngressProgressingCondition(conditions []operatorv1.OperatorCondition, ic *operatorv1.IngressController, service *corev1.Service, platform *configv1.PlatformStatus, deployment *appsv1.Deployment) operatorv1.OperatorCondition {
 	condition := operatorv1.OperatorCondition{
 		Type:   operatorv1.OperatorStatusTypeProgressing,
 		Status: operatorv1.ConditionFalse,
 	}
+	messages := []string{}
+	progressingReasons := []string{}
+
+	// If deployment doesn't have the desired number of replicas, mark as progressing.
+	wantReplicas := int32(0)
+	if deployment.Spec.Replicas != nil {
+		wantReplicas = *deployment.Spec.Replicas
+	}
+	haveReplicas := deployment.Status.Replicas
+	// First check if we have the total number of desired replicas, then check if they are all up-to-date.
+	if haveReplicas != wantReplicas {
+		progressingReasons = append(progressingReasons, "Reconciling")
+		messages = append(messages, fmt.Sprintf("Have %d available router pods, want %d.", haveReplicas, wantReplicas))
+	} else {
+		updatedReplicas := deployment.Status.UpdatedReplicas
```

Comment:
I figured there no need to check update == have if have != want. If Have != want, then update usually != want and doesn't really tell us anything new.